spyder/plugins/editor/widgets/tests/test_bookmarks.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+#
+
+"""
+Tests for bookmarks.
+"""
+
+# Standard library imports
+import os.path as osp
+
+try:
+    from unittest.mock import Mock
+except ImportError:
+    from mock import Mock  # Python 2
+
+# Third party imports
+import pytest
+from qtpy.QtGui import QTextCursor
+from qtpy.QtWidgets import QMainWindow
+
+# Local imports
+from spyder import version_info
+from spyder.py3compat import to_text_string
+from spyder.utils.qthelpers import qapplication
+import spyder.plugins.editor.widgets.codeeditor as codeeditor
+
+
+# --- Helper methods
+# -----------------------------------------------------------------------------
+def reset_emits(editor):
+    "Reset signal mocks."
+    editor.bookmarks_changed.reset_mock()
+
+
+def editor_assert_helper(editor, block=None, bm=None, emits=True):
+    """Run the tests for call to add_remove_breakpoint.
```

Comment:
Instead of duplicating this code from another test function (non-DRY and modular), you should make it a globally accessible helper function instead. See my [comment](https://github.com/spyder-ide/spyder/pull/8644#discussion_r251208258) on another PR for how to do that.